File: scripts/test_solutions/1899_merge_triplets.js
var mergeTriplets = function(triplets, target) {
    const good = new Set();
    for (const [a, b, c] of triplets) {
        if (a <= target[0] && b <= target[1] && c <= target[2]) {
            if (a === target[0]) good.add(0);
            if (b === target[1]) good.add(1);
            if (c === target[2]) good.add(2);
        }
    }
    return good.size === 3;
};

File: scripts/test_solutions/1899_merge_triplets.cpp
class Solution {
public:
    bool mergeTriplets(vector<vector<int>>& triplets, vector<int>& target) {
        set<int> good;
        for (auto& t : triplets) {
            if (t[0] <= target[0] && t[1] <= target[1] && t[2] <= target[2]) {
                if (t[0] == target[0]) good.insert(0);
                if (t[1] == target[1]) good.insert(1);
                if (t[2] == target[2]) good.insert(2);
            }
        }
        return good.size() == 3;
    }
};
